pingcap/tidb-operator
# deploy a discovery component for every DmCluster

### What problem does this PR solve?

closes #3906

### What is changed and how does it work?

deploy Discovery for every DmCluster.

### Code changes


### Tests


  1. Deploy an old version of TiDB-Operator
  2. Deploy a TidbCluster in ns-A
  3. Deploy a DmCluster in ns-A
  4. Upgrade TiDB-Operator to this PR's version
  5. Observe DMCluster in ns-A rolling upgraded with a Discovery
  6. Deploy a DmCluster in ns-B


### Side effects

   - DM-master will be rolling upgraded after upgrade TiDB-Operator to a version with this PR.

### Related changes


### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note
Support deploying DmCluster in a different namespace of a TidbCluster
NOTE: after TiDB Operator upgraded, all existing DM-master members in DmCluster will be rolling updated.
```
